entity-service-impl/src/test/java/org/hypertrace/entity/query/service/converter/ValueHelperTest.java

```diff
@@ -103,6 +104,18 @@ void testConvertStringValueToSubDocumentValue() throws ConversionException, IOEx
               Value.newBuilder().setString("Mars").setValueType(ValueType.STRING).build()));
     }
 
+    @Test
+    void testConvertMapValueToSubDocumentValue() throws ConversionException, IOException {
+      assertEquals(
+          SubDocumentValue.of(
+              new JSONDocument("{ \"key\": {\"value\": {\"string\": \"value\" }}}")),
+          valueHelper.convertToSubDocumentValue(
+              Value.newBuilder()
+                  .putAllStringMap(Map.of("key", "value"))
```

Comment:
Nit: Can you name `value` as something else (say `value1`)? It's a bit confusing which of the 2 values in the JSONDocument this `value` is referring to.